### PR TITLE
Document GetTattooShopDlcItemData

### PR DESCRIPTION
**Pull request by freedy69**
Researched the struct values from shop_tattoo.meta and changed few stuff.

### CODE REVIEW

File: FILES/GetTattooShopDlcItemData.md [hunk 1]
```diff
@@ -18,42 +20,28 @@ Character types:
...
-    ZONE_NONE = 7,
+	uint LockHash; // Hash that is to be used with IS_CONTENT_ITEM_LOCKED
+	int Index; // This is the unique ID of the tattoo, but sometimes can also be zero.
+	int Collection; // Which collection this tattoo belongs to
+	uint Preset; // Preset hash of this tattoo (AKA it's name, to be used with ADD_PED_DECORATION_FROM_HASHES)
+	int Cost; // How much this tattoo costs in shops
+	int eFacing; // Facing value of this tattoo
+	uint UpdateGroup; // Where exactly this tattoo is located on the ped. (E.G ARM_LEFT_FULL_SLEEVE.)
+	const char* NameTextLabel; // Name of this tattoo as a text label
 };
```
Comment (PsychoShock) [inline FILES/GetTattooShopDlcItemData.md:34]: Try to set a table with all this information.

File: FILES/GetTattooShopDlcItemData.md [hunk 2]
```diff
@@ -18,42 +20,28 @@ Character types:
...
 * **characterType**: Character types 0 = Michael, 1 = Franklin, 2 = Trevor, 3 = MPMale, 4 = MPFemale.
-* **decorationIndex**: Tattoo index, value between 0 and GetNumDecorations(characterType).
+* **decorationIndex**: Tattoo index, value between 0 and [GET_NUM_TATTOO_SHOP_DLC_ITEMS](#_0x278F76C3B0A8F109).
 * **outComponent**: The referenced struct.
 
 ## Return value
-A bool indicating that the tattoo data could be fetched(?)
+A bool indicating that the tattoo data exists in the files.
 
 ## Examples
```
Comment (PsychoShock) [inline FILES/GetTattooShopDlcItemData.md:46]: __If you can__, try adding a LUA example

**Review by PsychoShock — CHANGES_REQUESTED** [review]

File: FILES/GetTattooShopDlcItemData.md (hunk 1 above)
Comment (PsychoShock) [reply FILES/GetTattooShopDlcItemData.md:34]: You can check https://github.com/citizenfx/natives/pull/984#issuecomment-1936900707

File: FILES/GetTattooShopDlcItemData.md [hunk 3]
```diff
@@ -9,51 +9,37 @@ aliases: ["0xFF56381874F82086","_GET_TATTOO_COLLECTION_DATA"]
...
+| `uint`        |   Collection  |  Collection of the tattoo.                                             |
+| `uint`        |   Preset      |  Preset of this tattoo.                                                |
+| `int`         |   Cost        |  Cost of this tattoo in shops.                                         |
+| `int`         |   eFacing     |  Facing (Secondary placement value) of this tattoo.                    |
+| `uint`        |   UpdateGroup |  Location of this tattoo.                                              |
+| `const char*` | NameTextLabel |   This tattoo's name as a text label.                                  |
 
 
 ## Parameters
 * **characterType**: Character types 0 = Michael, 1 = Franklin, 2 = Trevor, 3 = MPMale, 4 = MPFemale.
```
Comment (PsychoShock) [inline FILES/GetTattooShopDlcItemData.md:37]: No need to have the character types information since it's stored on the table. You can just set `Character types`.

File: FILES/GetTattooShopDlcItemData.md [hunk 4]
```diff
@@ -67,4 +53,41 @@ if (Citizen.invokeNative("0xFF56381874F82086", characterType, tattooIndex, struc
...
+        Price = Price,
+        eFacing = eFacing,
+        UpdateGroup = UpdateGroup,
+        TextLabel = TextLabel
+    }
+end
 
+function GetTattooDlcItemDataTable(CharacterType, DecorationIndex)
+	local blob = string.rep('\0\0\0\0\0\0\0\0', 7+16)
+	if not Citizen.InvokeNative(0xFF56381874F82086, CharacterType, DecorationIndex, blob) then return nil end -- Data doesn't exist, return a nil
```
Comment (PsychoShock) [inline FILES/GetTattooShopDlcItemData.md:82]: Try avoiding this.

**Review by PsychoShock — CHANGES_REQUESTED** [review]

File: FILES/GetTattooShopDlcItemData.md (hunk 4 above)
Comment (freedy69) [reply FILES/GetTattooShopDlcItemData.md:82]: The nil return?

Comment (PsychoShock) [reply FILES/GetTattooShopDlcItemData.md:82]: Invoking a native

Comment (freedy69) [reply FILES/GetTattooShopDlcItemData.md:82]: Don't think there's a different way to do this though

Comment (AvarianKnight) [reply FILES/GetTattooShopDlcItemData.md:82]: There isn't.

File: FILES/GetTattooShopDlcItemData.md [hunk 5]
```diff
@@ -9,51 +9,37 @@ aliases: ["0xFF56381874F82086","_GET_TATTOO_COLLECTION_DATA"]
...
-0 = Michael,
-1 = Franklin,
-2 = Trevor,
-3 = MPMale,
-4 = MPFemale
-```
+Returns data that adheres to the tattoo shop item data that is used in shop_tattoo.meta
 
-```csharp
-enum TattooZoneData
```
Comment (AvarianKnight) [inline FILES/GetTattooShopDlcItemData.md:22]: Why was this removed?

**Review by AvarianKnight — CHANGES_REQUESTED** [review]

File: FILES/GetTattooShopDlcItemData.md (hunk 5 above)
Comment (freedy69) [reply FILES/GetTattooShopDlcItemData.md:22]: hey so i removed this since the tattoo zones don't have any use in this specific native (like, the struct doesn't return the zone or anything, nor do the parameters need it)

**Review by AvarianKnight — CHANGES_REQUESTED** [review]

File: FILES/GetTattooShopDlcItemData.md [hunk 6]
```diff
@@ -9,51 +9,43 @@ aliases: ["0xFF56381874F82086","_GET_TATTOO_COLLECTION_DATA"]
 BOOL GET_TATTOO_SHOP_DLC_ITEM_DATA(int characterType, int decorationIndex, Any* outComponent);
 ```
 
+Returns data that adheres to the tattoo shop item data that is used in shop_tattoo.meta
+
 Character types:
 ```
```
Comment (AvarianKnight) [inline FILES/GetTattooShopDlcItemData.md:15]:
```suggestion
```c
```

File: FILES/GetTattooShopDlcItemData.md [hunk 7]
```diff
@@ -9,51 +9,43 @@ aliases: ["0xFF56381874F82086","_GET_TATTOO_COLLECTION_DATA"]
...
+	TATTOO_SP_TREVOR = 2,
+	TATTOO_MP_FM = 3,
+	TATTOO_MP_FM_F = 4
+}
 ```
 
-```csharp
-enum TattooZoneData
+Returned struct properties:
+```
```
Comment (AvarianKnight) [inline FILES/GetTattooShopDlcItemData.md:27]:
```suggestion
```c
```
